python/ray/serve/_private/autoscaling_state.py

```diff
@@ -301,6 +314,86 @@ def get_total_num_requests(self) -> float:
         return total_requests
 
 
+class ApplicationAutoscalingState:
+    """Manages autoscaling for a single application."""
+
+    def __init__(
+        self,
+        app_name: ApplicationName,
+        deployment_autoscaling_states: Dict[DeploymentID, DeploymentAutoscalingState],
+    ):
+        self._app_name = app_name
+        self._deployment_autoscaling_states = deployment_autoscaling_states
+        self._config: ServeApplicationSchema = None
+        self._policy: AutoscalingPolicy = None
+        self._policy_state: Dict[str, Dict] = None
+
+    def register(
+        self,
+        config: ServeApplicationSchema,
+    ):
+        self._config = config
+        self._policy = self._config.get_autoscaling_policy()
+        self._policy_state = {}
+
+    def get_scaling_decisions(
+        self, deployments: Dict[str, DeploymentDetails], _skip_bound_check: bool = False
+    ) -> Dict[str, int]:
+        autoscaling_contexts: Dict[str, AutoscalingContext] = {}
+        decisions: Dict[str, int] = {}
+        for name, deployment_detail in deployments.items():
+            deployment_id: DeploymentID = DeploymentID(
+                name=name, app_name=self._app_name
+            )
+            deployment_autoscaling_state = self._deployment_autoscaling_states[
+                deployment_id
+            ]
+            autoscaling_context: AutoscalingContext = AutoscalingContext(
```

Comment:
instead of constructing the `autoscaling_context` here, can we instead create a function in `deployment_autoscaling_state` to `get_autoscaling_context()`